tests/attacks/property_inference_test.py

```diff
@@ -5,22 +5,88 @@
     new_dataset_from_size_dict,
 )
 from privacy_evaluator.utils.trainer import trainer
-from privacy_evaluator.models.tf.cnn import ConvNet
+from privacy_evaluator.models.torch.cnn import ConvNet
+from typing import Dict, List
+from torch import nn
 
+# ratio for target model
+NUM_ELEMENTS_PER_CLASSES = {0: 1000, 1: 1000}
+# dataset for the attack
+DATASET = "MNIST"
+# number of channels for CNN
+NUM_CHANNELS = (1, 16, 32, 64)
+# number of epochs for trainer
+NUM_EPOCHS = 2
+# count of shadow training sets, must be even
+AMOUNT_SETS = 2
+# ratio and size for unbalanced data sets
+SIZE_SET = 100
+# ratios for different properties in sub-attacks
+RATIOS_FOR_ATTACK = [0.9, 0.3]
+# classes the attack should be performed on
+CLASSES = [4, 5]
+# 0: no information; 1: backbone (most important) information; 2: utterly detailed
+VERBOSE = 1
 
-def test_property_inference_attack():
-    train_dataset, test_dataset = dataset_downloader("CIFAR10")
+
+def test_property_inference_attack(
+    num_elements_per_classes: Dict[int, int] = NUM_ELEMENTS_PER_CLASSES,
+    dataset: str = DATASET,
+    num_channels: int = NUM_CHANNELS,
+    num_epochs: int = NUM_EPOCHS,
+    amount_sets: int = AMOUNT_SETS,
+    size_set: int = SIZE_SET,
+    ratios_for_attack: List[float] = RATIOS_FOR_ATTACK,
+    classes: List[int] = CLASSES,
+    verbose: int = VERBOSE,
+):
+    train_dataset, test_dataset = dataset_downloader(dataset)
     input_shape = test_dataset[0][0].shape
-    num_elements_per_classes = {0: 1000, 1: 1000}
+
     num_classes = len(num_elements_per_classes)
 
     train_set = new_dataset_from_size_dict(train_dataset, num_elements_per_classes)
+    # num_channels and input_shape are optional in cnn.py
+    model = ConvNet(num_classes, input_shape, num_channels=num_channels)
 
-    model = ConvNet(num_classes, input_shape)
-    trainer(train_set, num_elements_per_classes, model)
+    print("Start training target model ...\n")
```

Comment:
Should this print statement not depend on what is set in verbose?

Reply:
fixed